packages/xo-web-6/src/App/Visualization/hostStats.tsx

```diff
@@ -0,0 +1,556 @@
+import React, { Component } from 'react'
+import {
+  XAxis,
+  YAxis,
+  AreaChart,
+  Tooltip,
+  Legend,
+  Area,
+  Brush,
+  CartesianGrid,
+} from 'recharts'
+
+import Xo from 'xo-lib'
+import moment, { max } from 'moment'
+const NB_VALUES = 118
+
+const xo = new Xo({ url: '/' })
+xo.open().then(() => xo.signIn({ email: 'admin@admin.net', password: 'admin' }))
+const signedIn = new Promise(resolve => xo.once('authenticated', resolve))
+const xoCall = (method:string, params:object) => signedIn.then(() => xo.call(method, params))
+
+const getObject = (id: any) =>
+  xoCall('xo.getAllObjects', { filter: { id } }).then(objects => objects[id])
+
+export default class Visualization extends Component<any, any> {
+  state: any = {
+    granularity: 'seconds',
+    format: 'LTS',
+
+    //  Host
+    //  cpu
+    cpuDataHost: [],
+    cpusHost: [],
+
+    // Memory
+    memoryDataHost: [],
+    memoryMaxHost: 0,
+
+    //  Network
+    networkDataHost: [],
+    networksRxHost: [],
+    networksTransmissionHost: [],
+    networksReceptionHost: [],
+
+    //  Load
+    loadDataHost: [],
+    maxLoad: 0,
+  }
+
+  componentDidMount() {
+    setInterval(this.fetchHostStats.bind(this), 5e3)
+  }
+
+  fetchHostStats = () => {
+    getObject('470b64a4-2767-4e1d-a20c-fbc2a6d4de57').then((host: any) => {
+      this.setState({ memoryMaxHost: host.memory.size })
+    })
+   
+    xoCall('host.stats', {
+      host: '470b64a4-2767-4e1d-a20c-fbc2a6d4de57',
+      granularity: this.state.granularity,
+    }).then(
+      ({
+        endTimestamp,
+        stats: { cpus },
+        interval,
+        stats: { load },
+        stats: { memory, memoryFree = memory },
+        stats: { pifs },
+      }) => {
+        let format: any
+        if (interval === 5 || interval === 60) {
+          format = 'LTS'
+        } else if (interval === 86400 || interval === 3600) {
+          format = 'l'
+        }
+
+        this.setState({ cpusHost: Object.keys(cpus) })
+        this.setState({ networksRxHost: Object.keys(pifs.rx) })
+        this.setState({ networksTransmissionHost: Object.keys(pifs.tx) })
+
+        let cpuDataHost: any[] = []
+        let memoryDataHost: any[] = []
+        let networkDataHost: any[] = []
+        let loadDataHost: any[] = []
+        let newDataMemory: any = []
+  
+          newDataMemory = memoryFree.map(
+            (value: any, index: any) => memory[index] - value
+          )
+        
+        for (var i = 0; i < NB_VALUES; i++) {
```

Comment:
This is what you're doing:
```js
const rawData = {
  cpu: [ ... ],
  memory: [ ... ]
}
const computedData = {
  cpu: [],
  memory: []
}
for (let i = 0; i < NB_VALUES; i++) {
  computedData.cpu[i] = transform(rawData.cpu[i])
  computedData.memory[i] = transform(rawData.memory[i])
}
```
But it would make more sense the other way around:
```js
const rawData = {
  cpu: [ ... ],
  memory: [ ... ]
}
const computedData = {}
computedData.cpu = rawData.cpu.forEach(value => transform(value))
computedData.memory = rawData.memory.forEach(value => transform(value))
```